contracts/traderjoe/FarmLens.sol

```diff
@@ -0,0 +1,199 @@
+    address public wavax; // 0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7;
+    IJoeFactory public joeFactory; // IJoeFactory(0x9Ad6C38BE94206cA50bb0d90783181662f0Cfa10);
+    IMasterChef public chef; //0xd6a4F121CA35509aF06A0Be99093d08462f53052
+    IMasterChef public chefv3; //0x188bED1968b795d5c9022F6a0bb5931Ac4c18F00
+
+    constructor(
+        address joe_,
+        address wavax_,
+        IJoeFactory joeFactory_,
+        IMasterChef chef_,
+        IMasterChef chefv3_
+    ) public {
+        joe = joe_;
+        wavax = wavax_;
+        joeFactory = IJoeFactory(joeFactory_);
+        chef = chef_;
+        chefv3 = chefv3_;
+    }
+
+    function getAvaxPrice() public view returns (uint256) {
+        uint256 priceFromWavaxUsdt = _getAvaxPrice(IJoePair(address(0xeD8CBD9F0cE3C6986b22002F03c6475CEb7a6256))); // 18
+        uint256 priceFromWavaxUsdc = _getAvaxPrice(IJoePair(address(0x87Dee1cC9FFd464B79e058ba20387c1984aed86a))); // 18
+        uint256 priceFromWavaxDai = _getAvaxPrice(IJoePair(address(0xA389f9430876455C36478DeEa9769B7Ca4E3DDB1))); // 18
+
+        uint256 sumPrice = priceFromWavaxUsdt.add(priceFromWavaxUsdc).add(priceFromWavaxDai); // 18
+        uint256 avaxPrice = sumPrice / uint256(3); // 18
+        return avaxPrice; // 18
+    }
+
+    function _getAvaxPrice(IJoePair pair) public view returns (uint256) {
+        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves();
+
+        if (pair.token0() == wavax) {
+            reserve1 = reserve1.mul(_tokenDecimalsMultiplier(pair.token1())); // 18
+            return (reserve1.mul(uint256(1e18))) / reserve0; // 18
+        } else {
+            reserve0 = reserve0.mul(_tokenDecimalsMultiplier(pair.token0())); // 18
+            return (reserve0.mul(uint256(1e18))) / reserve1; // 18
+        }
+    }
+
+    function getPriceInUSD(address tokenAddress) public view returns (uint256) {
+        return getAvaxPrice() / getPriceInAvax(tokenAddress); // 18
+    }
+
+    // Need to be aware of decimals here, not always 18, it depends on the token
+    function getPriceInAvax(address tokenAddress) public view returns (uint256) {
+        if (tokenAddress == wavax) {
+            return uint256(1e18);
+        }
+
+        IJoePair pair = IJoePair(joeFactory.getPair(tokenAddress, wavax));
+
+        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves();
+        address token0Address = pair.token0();
+        address token1Address = pair.token1();
+
+        if (token0Address == wavax) {
+            reserve1 = reserve1.mul(_tokenDecimalsMultiplier(token1Address)); // 18
+            return (reserve1.mul(uint256(1e18))) / reserve0; // 18
+        } else {
+            reserve0 = reserve0.mul(_tokenDecimalsMultiplier(token0Address)); // 18
+            return (reserve0.mul(uint256(1e18))) / reserve1; // 18
+        }
+    }
+
+    function _tokenDecimalsMultiplier(address tokenAddress) public pure returns (uint256) {
```

Comment:
Is this function supposed to be private? (I assumed so cos it was prefixed with _)